.github/workflows/Trgr InitDev workflows in Implementation repo.yml

```diff
@@ -57,8 +57,34 @@ jobs:
           $current_branch = '${{env.HEAD_REF}}'
         }
         Write-Host "Current Branch: $current_branch"   
-        echo "current_branch=$current_branch" >> $Env:GITHUB_ENV                   
+        echo "current_branch=$current_branch" >> $Env:GITHUB_ENV
+    - name: Check Ed-Fi-ODS-Implementation Repo PR exists or not
+      shell: pwsh
+      run: |
+        $accessToken = "${{ env.EDFI_ODS_IMP_TOKEN  }}"
+        $repositoryName ="Ed-Fi-ODS-Implementation"
+        $apiUrl = "https://api.github.com/repos/${{env.REPOSITORY_OWNER}}/$repositoryName/pulls?state=open&head=${{env.REPOSITORY_OWNER}}:${{env.current_branch}}"
+        $headers = @{
+            Authorization = "Bearer $accessToken"
+            Accept = "application/vnd.github.v3+json"
+        }
+        $response = Invoke-RestMethod -Uri $apiUrl -Headers $headers
+        # Access x-ratelimit-limit and x-ratelimit-remaining from response headers
+        $rateLimitLimit = $response.Headers['x-ratelimit-limit']
+        $rateLimitRemaining = $response.Headers['x-ratelimit-remaining']
+        Write-Host "x-ratelimit-limit: $rateLimitLimit"
+        Write-Host "x-ratelimit-remaining: $rateLimitRemaining"              
+        foreach ($pr in $response) {
+         Write-Host "current branch: ${{env.current_branch}}"
+         Write-Host "ref: $($pr.head.ref)"
+          if ($pr.head.ref -eq "${{ env.current_branch }}") 
+          {
+            Write-Host "PR #$($pr.number): $($pr.head.label)"
+            echo "EXIT_STEP=true" >> $Env:GITHUB_ENV
```

Comment:
I think that I follow the logic here. If we find a match, then `EXIT_STEP == true` and the steps below will run. If no branch match found, then the steps won't run.

Suggestion: write a [formatted error message](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message) after the foreach loop if `EXIT_STEP != true`. This will make it easier to identify why the problem occurred. It creates an annotation that is very visible in the action results, and the message will be formatted in red in the log.

Final suggestion: `echo` is okay here, but it is a little strange to me seeing `Write-Host` above and then `echo` here. Wouldn't it be better for this to remain consistent?

Reply:
why I used echo for setting environment variable in PowerShell echo "EXIT_STEP=true">> $env:GITHUB_ENV https://www.jamescroft.co.uk/setting-github-actions-environment-variables-in-powershell/